src/rollback_to_stable/rts_btree.c

```diff
@@ -735,8 +739,8 @@ __rts_btree_abort_ondisk_kv(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
          * windows can be the same. To abort these updates, check for any stable update from history
          * store or remove the key.
          */
-        if (tw->start_ts == tw->stop_ts && tw->durable_start_ts == tw->durable_stop_ts &&
-          tw->start_txn == tw->stop_txn) {
+        if (WT_TIME_WINDOW_HAS_STOP_PREPARE(tw) && tw->start_prepared_id == tw->stop_prepared_id &&
```

Comment:
As discussed in the previous pr. We should not check this WT_TIME_WINDOW_HAS_STOP_PREPARE. We should check whether prepare ts, start ts, durable ts, and txn are all equal